models/user.model.js

```diff
@@ -35,6 +39,21 @@ const userSchema = new Schema(
   { timestamps: true },
 );
 
+userSchema.pre('save', function (next) {
+  if (!this.isModified('password')) return next();
+
+  bcrypt.genSalt(SALT_WORK_FACTOR, (err, salt) => {
+    if (err) return next(err);
+
+    bcrypt.hash(this.password, salt, (err, hash) => {
+      if (err) return next(err);
+
+      this.password = hash;
+      next();
+    });
+  });
+});
+
```

Comment:
```suggestion
userSchema.pre('save', async function (next) {
  const user = this;
  
  if (user.isModified('password')) {
    user.password = await bcrypt.hash(user.password, SALT_WORK_FACTOR);
  }
  
  next();
});

```
- Đoạn này em nên dùng async await cho gắn gọn
- Dùng phủ định tùy trường hợp giúp middleware pre này sau có thể bắt thêm trường.